src/js/components/DataTable/Header.js

```diff
@@ -182,26 +189,46 @@ const Header = forwardRef(
               {onSelect && (
                 <CheckBox
                   a11yTitle={
-                    selected.length === data.length
+                    totalSelected === data.length
                       ? 'unselect all'
                       : 'select all'
                   }
                   checked={
-                    selected.length > 0 &&
-                    data.length > 0 &&
-                    selected.length === data.length
+                    groupBy?.select ?
```

Comment:
In the `onUpdate` story, I'm not able to click the checkbox in the datatable header to select all the data.

Reply:
Good catch. I somehow broke this (unitialized variable.)